File: GUISM_SERVER/source/descendencia/Descendencia.java
package descendencia;

import habilidade.Habilidade;
import habito.Habito;
import kikaha.jdbi.serializers.Column;
import kikaha.jdbi.serializers.Entity;
import lombok.Data;
import sanidade.Sanidade;
import situacao.Situacao;

import java.util.Set;

@Data
@Entity
public class Descendencia {

    @Column
    Long idDescendencia;

    @Column
    String nomeDescendencia;

    @Column
    String descDescendencia;

    Set<Habilidade> habilidades;

    Set<Habito> habitos;

    Sanidade sanidade;

    Set<Situacao> situacoes;
}

File: src/components/descendencia/Descendencia.ts
import Habilidade from "../habilidade/Habilidade";
import Situacao from "../situacao/situacao";
import Habito from "../habito/Habito";
import Sanidade from "../sanidade/Sanidade";


export interface Descendencia {
    idDescendencia: number;
    nomeDescendencia: string;
    descDescendencia: string;
    habilidades: Habilidade[];
    habitos: Habito[];
    sanidade: Sanidade;
    situacoes: Situacao[];
}
